hapi-fhir-structures-r4/src/test/java/ca/uhn/fhir/parser/JsonParserR4Test.java

```diff
@@ -1490,6 +1491,38 @@ public void testNestedModifierExtensions() {
 		assertThat(patientStringXML).containsOnlyOnce("<modifierExtension");
 	}
 
+	// this test is to ensure contained resources that are referenced by the
+	// entire resource will still be able to be processed in a transaction
+	@Test
+	public void encodeResourceToString_withContainedResource_shouldAddContainedReferenceIdEvenIfNotPresent() {
+		// setup
+		Patient patient = new Patient();
+		{
+			Practitioner practitioner = new Practitioner();
+			practitioner.addName()
+				 .setFamily("Hibert");
+			patient.addName()
+				 .setFamily("Simpson");
+			patient.getContained().add(practitioner);
+			patient.addGeneralPractitioner(new Reference(practitioner));
+		}
+
+		IParser parser = ourCtx.newJsonParser();
+
+		// test
+		String encoded = parser.encodeResourceToString(patient);
+		ourLog.info(encoded);
+
+		// verify
+		assertNotNull(encoded);
+		assertTrue(isNotBlank(encoded));
+		assertFalse(patient.getGeneralPractitioner().isEmpty());
```

Comment:
Nitpick, non-blocking: asserting false on isEmpty() is kind of hard to read
```suggestion
		assertThat(patient.getGeneralPractitioner()).isNotEmpty();
```